webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/Compliance.scala

```diff
@@ -131,27 +137,143 @@ final case class ByRuleDirectiveCompliance(
 sealed trait ByRuleComponentCompliance {
   def name:       String
   def compliance: ComplianceLevel
+
+  def getValues(predicate: ByRuleNodeCompliance => Boolean): Seq[ByRuleNodeCompliance]
+
+  def withFilteredElement(predicate: ByRuleNodeCompliance => Boolean): Option[ByRuleComponentCompliance]
+
+  def componentValues: List[ByRuleNodeCompliance]
+
+  def componentValues(v: String): List[ByRuleNodeCompliance] = componentValues.filter(_.values.exists(_.componentValue == v))
+
+  def status: ReportType
```

Comment:
the meaning of status here is "the worst report type in sub component, based on target rules?"

Reply:
In fact i don't think it's used by component but i may be wrong, but for value it's is the worst in all cases. Probably in a block it should depends on reporting logic, But I'm not sure it is really used for a block
.